pkg/harvester/mixins/harvester-vm/index.js

```diff
@@ -404,8 +405,7 @@ export default {
             minExponent: 3,
           });
 
-          const allVolumeStatus = JSON.parse(vm.metadata?.annotations?.[HCI_ANNOTATIONS.VM_VOLUME_STATUS] || '[]');
-          const volumeStatus = allVolumeStatus.find(volume => realName === volume.name);
+          const volumeStatus = this.pvcs.find(P => P.metadata?.name === volumeName)?.relatedPV?.metadata?.annotations?.[HCI_ANNOTATIONS.VOLUME_ERROR];
```

Comment:
I have a question about what if there have two PVCs: they have the same name but have a different namespace. It might find a name that does not match.

Reply:
Use pvc.id to filter pvc list now.